.github/container/manifest.yaml

```diff
@@ -0,0 +1,103 @@
+# Updated in: 2023-12-01
+jax:
+  url: https://github.com/google/jax.git
+  tracking_ref: main
+  latest_verified_commit: 595117b70c11055e569480b80907d8c8a9901805
+  mode: git-clone
+xla:
+  url: https://github.com/openxla/xla.git
+  tracking_ref: main
+  latest_verified_commit: 78a5297d8e4301cb3ba2514061f56f89104e3d88
+  mode: git-clone
+flax:
+  url: https://github.com/google/flax.git
+  mirror_url: https://github.com/nvjax-svc-0/flax.git
+  tracking_ref: main
+  latest_verified_commit: 230b0d77e98da22b6e574c3cbff743ca1504bfca
+  mode: git-clone
+  patches:
+    pull/3340/head: file://patches/flax/PR-3340.patch # Add Sharding Annotations to Flax Modules
+transformer-engine:
+  url: https://github.com/NVIDIA/TransformerEngine.git
+  tracking_ref: main
+  latest_verified_commit: 92c1e500dd14608e54f75df8276baa1104c61d48
+  mode: git-clone
+t5x:
+  url: https://github.com/google-research/t5x.git
+  mirror_url: https://github.com/nvjax-svc-0/t5x.git
+  tracking_ref: main
+  latest_verified_commit: 1bfd2f15e5e77b09d60301367f67fdc9bb756b46
+  mode: git-clone
+  patches:
+    mirror/patch/partial-checkpoint-restore: file://patches/t5x/mirror-patch-partial-checkpoint-restore.patch # pull/1392/head  # https://github.com/google-research/t5x/pull/1392: Add support for partial checkpoint restore
+    mirror/patch/dali-support: file://patches/t5x/mirror-patch-dali-support.patch # pull/1393/head  # https://github.com/google-research/t5x/pull/1393: Adds DALI support to t5x
+    mirror/patch/t5x_te_in_contrib_noindent: file://patches/t5x/mirror-patch-t5x_te_in_contrib_noindent.patch # pull/1391/head  # https://github.com/google-research/t5x/pull/1391: Adds transformer engine support and GPU optimizations to T5x (enables H100)
+paxml:
+  url: https://github.com/google/paxml.git
+  mirror_url: https://github.com/nvjax-svc-0/paxml.git
+  tracking_ref: main
+  latest_verified_commit: 7ae682d4d99630008e190b96c5296990297175c2
+  mode: git-clone
+  patches:
+    pull/46/head: file://patches/paxml/PR-46.patch # adds Transformer Engine support
+praxis:
+  url: https://github.com/google/praxis.git
+  mirror_url: https://github.com/nvjax-svc-0/praxis.git
+  tracking_ref: main
+  latest_verified_commit: b6f32fa0fc6721db1cec75972b0f569c82095956
+  mode: git-clone
+  patches:
+    pull/27/head: file://patches/praxis/PR-27.patch # This PR allows XLA:GPU to detect the MHA pattern more easily to call fused kernels from cublas.
+    pull/36/head: file://patches/praxis/PR-36.patch # adds Transformer Engine support
+lingvo:
+  # Used only in ARM pax builds
+  url: https://github.com/tensorflow/lingvo.git
+  tracking_ref: master
+  latest_verified_commit: 0274fa20b4ff194c1c118b94b5f778caa5d9a84a
+  mode: git-clone
+tensorflow-text:
+  # Used only in ARM pax builds
+  url: https://github.com/tensorflow/text.git
+  tracking_ref: v2.13.0
+  latest_verified_commit: 917a681d7220ebf9b62a08b6f9ce7b7db886ddef
+  mode: git-clone
+pydantic:
+  version: X.Y.Z
```

Comment:
Actually I planned on including this pin for the ARM build for pax, but I see that you removed that in the pip-compile change: https://github.com/NVIDIA/JAX-Toolbox/pull/371/files#diff-2f69dd1b68217a122ece65bac9df3f6763e53ca122f367ce90c6610ca21584e9L41

Did you happen to test that this is no longer needed? 

I didn't actually implement the pip-constraint mechanism in this PR since I didn't have an example, so if you've tested that we don't need this anymore, I can remove it for now. If you haven't, then I can implement it. I think given how pinning appears to be platform specific, I was thinking of something like this:
```yaml
pydantic:
  versions:
     arm64: ==1.10.13
  mode: pip-constraint
```

Reply:
IIRC pydantic was blocking the installation of T5X because, back then, JAX requested a version that was too new for T5X and pip somehow did not want to downgrade. The issue doesn't seem to exist anymore once we moved to pip-compile.